DSharpPlus/Clients/DiscordClient.Dispatch.cs

```diff
@@ -601,6 +602,87 @@ internal async Task HandleDispatchAsync(GatewayPayload payload)
 
     #endregion
 
+    #region Webhook Events
+
+    private async Task ReceiveWebhookEventsAsync()
+    {
+        while (!this.webhookEventReader.Completion.IsCompleted)
+        {
+            DiscordWebhookEvent payload = await this.webhookEventReader.ReadAsync();
+
+            try
+            {
+                await HandleWebhookDispatchAsync(payload);
+            }
+            catch (Exception ex)
+            {
+                this.Logger.LogError(ex, "Dispatch threw an exception: ");
+            }
+        }
+    }
+
+    private async Task ReceiveInteractionEventsAsync()
+    {
+        while (!this.interactionEventReader.Completion.IsCompleted)
+        {
+            DiscordHttpInteractionPayload payload = await this.interactionEventReader.ReadAsync();
+            DiscordHttpInteraction interaction = payload.ProtoInteraction;
+
+            ulong? guildId = interaction.GuildId;
+            ulong channelId = interaction.ChannelId;
+
+            JToken rawMember = payload.Data["member"];
+            TransportMember? transportMember = null;
+            TransportUser transportUser;
+            if (rawMember != null)
+            {
+                transportMember = payload.Data["member"].ToDiscordObject<TransportMember>();
+                transportUser = transportMember.User;
+            }
+            else
+            {
+                transportUser = payload.Data["user"].ToDiscordObject<TransportUser>();
+            }
+
+            DiscordChannel channel = interaction.Channel;
+            channel.Discord = this;
+
+            await OnInteractionCreateAsync(guildId, channelId, transportUser, transportMember, channel, interaction);
+        }
+    }
+
+    private Task HandleWebhookDispatchAsync(DiscordWebhookEvent @event)
+    {
+        if (@event.ApplicationID != this.CurrentApplication.Id)
+        {
+            this.Logger.LogCritical

```

Comment:
Error, not critical. Critical should only be used when it affects the lifetime of the application

Reply:
this affects the integrity of the application, imo it should tear the application down entirely but that's out of scope for the library to do